worker/caasmodelconfigmanager/worker.go

```diff
@@ -73,96 +91,100 @@ func (config Config) Validate() error {
 	if config.RegistryFunc == nil {
 		return errors.NotValidf("RegistryFunc is missing")
 	}
+	if config.ImageRepoFunc == nil {
+		return errors.NotValidf("ImageRepoFunc is missing")
+	}
 	return nil
 }
 
 type manager struct {
-	catacomb catacomb.Catacomb
```

Comment:
Sorry to be a PITA, can we please keep the catacomb. I need it for my PR, due to introducing a watcher.